src/util/psbt/map/input.rs

```diff
@@ -150,10 +152,35 @@ pub struct Input {
 /// directly which signature hash type the user is dealing with. Therefore, the user is responsible
 /// for converting to/from [`PsbtSigHashType`] from/to the desired signature hash type they need.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct PsbtSigHashType {
     pub (in ::util::psbt) inner: u32,
 }
+serde_string_impl!(PsbtSigHashType, "a PsbtSigHashType data");
+
+impl fmt::Display for PsbtSigHashType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.ecdsa_hash_ty() {
+            Ok(ecdsa_sighash_ty) => fmt::Display::fmt(&ecdsa_sighash_ty, f),
+            Err(_) if self.inner <= 0xFF => write!(f, "{:#04x}", self.inner),
+            Err(_) => write!(f, "{:#010x}", self.inner),
+        }
+    }
+}
+
+impl FromStr for PsbtSigHashType {
```

Comment:
How about like this?
```
impl FromStr for PsbtSigHashType {
    type Err = SigHashTypeParseError;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        // We accept strings of form: "SIGHASH_ALL".
        if let Ok(ty) = EcdsaSigHashType::from_str(s) {
            return Ok(ty.into());
        }

        // We accept non-standard sighash values.
        // TODO: Use `trim_start_matches` once MSRV >= 1.30.
        if let Ok(inner) = u32::from_str_radix(s.trim_left_matches("0x"), 16) {
            return Ok(PsbtSigHashType { inner });
        }

        Err(SigHashTypeParseError::from(s.to_owned()))
    }
}
```

Reply:
Requires also
```

impl From<String> for SigHashTypeParseError {
    fn from(string: String) -> SigHashTypeParseError {
        SigHashTypeParseError { string }
    }
}
```

Reply:
I do not like the conversion into a specific error type from arbitrary `String`

Reply:
Fair point, I like what you've done. That means my 're-work sighash conversion methods' [PR](psbt/map/) is wrong. I'm going to pull out the patches that conflict with this PR into a separate PR to make it easier to merge everything and also I'm going to do as you do and make the inner field public.